src/Microsoft.TestPlatform.CrossPlatEngine/Adapter/FrameworkHandle.cs

```diff
@@ -110,6 +110,30 @@ public int LaunchProcessWithDebuggerAttached(string filePath, string workingDire
             return this.testRunEventsHandler.LaunchProcessWithDebuggerAttached(processInfo);
         }
 
+        /// <inheritdoc />
+        public bool AttachDebuggerToProcess(int pid)
+        {
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException("PID cannot be negative.");
+            }
+            if (pid == 0)
+            {
+                EqtTrace.Warning("Should not attach to process with id 0.");
```

Comment:
this is vague, and not true, you are free to attach to process id 0, if you don't want to permit it then throw erorr. Please prefix this message with the class and method name. and word it like a warning. e.g. "Attaching to a process with processId 0, is this intended?" and allow the debugger to attach, it will probably fail with access denied, but might not on non-windows / non-linux systems.

Reply:
Done.